ledger-tool/src/main.rs

```diff
@@ -3902,44 +3968,26 @@ fn main() {
                 );
                 let num_slots = value_t_or_exit!(arg_matches, "num_slots", usize);
                 let exclude_vote_only_slots = arg_matches.is_present("exclude_vote_only_slots");
-
-                let slots_iter = blockstore
-                    .reversed_optimistic_slots_iterator()
-                    .expect("Failed to get reversed optimistic slots iterator")
-                    .map(|(slot, hash, timestamp)| {
-                        let (entries, _, _) = blockstore
-                            .get_slot_entries_with_shred_info(slot, 0, false)
-                            .expect("Failed to get slot entries");
-                        let contains_nonvote = entries
-                            .iter()
-                            .flat_map(|entry| entry.transactions.iter())
-                            .flat_map(get_program_ids)
-                            .any(|program_id| *program_id != solana_vote_program::id());
-                        (slot, hash, timestamp, contains_nonvote)
-                    });
-
-                let slots: Vec<_> = if exclude_vote_only_slots {
-                    slots_iter
-                        .filter(|(_, _, _, contains_nonvote)| *contains_nonvote)
-                        .take(num_slots)
-                        .collect()
-                } else {
-                    slots_iter.take(num_slots).collect()
-                };
+                let slots =
+                    get_latest_optimistic_slots(&blockstore, num_slots, exclude_vote_only_slots);
 
                 println!(
                     "{:>20} {:>44} {:>32} {:>13}",
                     "Slot", "Hash", "Timestamp", "Vote Only?"
                 );
-                for (slot, hash, timestamp, contains_nonvote) in slots.iter() {
-                    let time_str = {
-                        let secs: u64 = (timestamp / 1_000) as u64;
-                        let nanos: u32 = ((timestamp % 1_000) * 1_000_000) as u32;
-                        let t = UNIX_EPOCH + Duration::new(secs, nanos);
-                        let datetime: DateTime<Utc> = t.into();
-                        datetime.to_rfc3339()
-                    };
-                    let hash_str = format!("{hash}");
+                for (slot, hash_and_timestamp_opt, contains_nonvote) in slots.iter() {
+                    let (time_str, hash_str) =
+                        if let Some((hash, timestamp)) = hash_and_timestamp_opt {
+                            let secs: u64 = (timestamp / 1_000) as u64;
+                            let nanos: u32 = ((timestamp % 1_000) * 1_000_000) as u32;
+                            let t = UNIX_EPOCH + Duration::new(secs, nanos);
+                            let datetime: DateTime<Utc> = t.into();
+
+                            (datetime.to_rfc3339(), format!("{hash}"))
+                        } else {
+                            let na = "N/A";
```

Comment:
Would `Undefined` or `Unknown` be more accurate than `N/A`?

Reply:
Yeah, I like `Unknown`